docker/buildx.sh

```diff
@@ -21,59 +21,65 @@ EXAMPLES
 REFERENCE
     https://github.com/openziti/ziti-tunnel-sdk-c/blob/main/docker/BUILD.md
 EOF
-    [[ $# -eq 1 ]] && {
-        return $1
-    } || {
+    if [[ $# -eq 1 ]]; then
+        return "$1"
+    else
         return 0
-    }
+    fi
 }
 
-BASENAME=$(basename $0) || exit $?
-DIRNAME=$(dirname $0) || exit $?
+#BASENAME=$(basename $0) || exit $?
+DIRNAME=$(dirname "$0") || exit $?
 
-while getopts :r:c OPT;do
+while getopts :r:chl OPT;do
     case $OPT in
         r)  CONTAINER_REPO=$OPTARG 
             ;;
         c) 	FLAGS+=$OPT     # don't checkout vZITI_VERSION
             ;;
         h) _usage; exit 0   # not an error
             ;;
-        *|\?) _usage 1      # error
+        l)  FLAGS+=$OPT     # also tag and push latest
+            ;;
+        \?|*) _usage 1      # error
             ;;
     esac
 done
 shift "$((OPTIND-1))"
 
 # default to latest
-: ${ZITI_VERSION:=$(git fetch --quiet --tags && git tag -l|sort -Vr|head -1|sed -E 's/^v(.*)/\1/')}
+: "${ZITI_VERSION:=$(git fetch --quiet --tags && git tag -l|sort -Vr|head -1|sed -E 's/^v(.*)/\1/')}"
 
 # required opts
-[[ ! -z ${CONTAINER_REPO:+true} ]] || {
+if [[ -z "${CONTAINER_REPO:-}" ]]; then
     echo "ERROR: missing -r REPO option to define container image repository name for image push" >&2
     _usage; exit 1
-}
+else
+    TAG_PARAMS="--tag=\"${CONTAINER_REPO}:${ZITI_VERSION}\""
+fi
 
 if [[ ${FLAGS:-} =~ c ]]; then
     echo "WARN: not checking out Git tag v${ZITI_VERSION}"
 else
     git diff --exit-code # bail if unstaged differences
     git fetch --tags
-    git checkout v${ZITI_VERSION}
+    git checkout "v${ZITI_VERSION}"
 fi
 
-docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
-egrep -q enabled /proc/sys/fs/binfmt_misc/qemu-arm
-docker run --rm arm64v8/alpine uname -a|egrep -q 'aarch64 Linux'
-docker run --rm arm32v7/alpine uname -a|egrep -q 'armv7l Linux'
+if [[ ${FLAGS:-} =~ l ]]; then
+    TAG_PARAMS+=" --tag=\"${CONTAINER_REPO}:latest\""
+fi
 
+docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
+grep -E -q 'enabled' /proc/sys/fs/binfmt_misc/qemu-arm
+docker run --rm --platform linux/arm64/v8 arm64v8/alpine uname -a |grep -Eq 'aarch64 Linux'
+docker run --rm --platform linux/arm/v7 arm32v7/alpine uname -a|grep -Eq 'armv7l Linux'
 docker buildx create --use --name=ziti-builder 2>/dev/null || docker buildx use --default ziti-builder
 
-docker buildx build $DIRNAME \
-    --platform linux/amd64,linux/arm/v7,linux/arm64 \
-    --build-arg ZITI_VERSION=${ZITI_VERSION} \
-    --tag ${CONTAINER_REPO}:${ZITI_VERSION} \
-    --tag ${CONTAINER_REPO}:latest \
+eval docker buildx build "${DIRNAME}" \
+    --platform="linux/amd64,linux/arm/v7,linux/arm64" \
+    --build-arg=ZITI_VERSION="${ZITI_VERSION}" \
+    "${TAG_PARAMS}" \
```

Comment:
Oh, maybe `eval` is eating the quotes that surround TAG_PARAMS so multiple tags can be passed successfully? Could we instead not use `eval` here and omit the quotes around TAG_PARAMS?

Reply:
The dockerfile change introduces a lot of duplication. The diff is between the two files is very small. In fact the only differences are the arguments to `COPY`.

    $ diff ~/tmp/dockerfile.fetch ~/tmp/dockerfile.copy
    17,19d16
    < COPY fetch-github-releases.sh .
    < RUN bash ./fetch-github-releases.sh ziti-edge-tunnel
    < 
    36d32
    < COPY --from=fetch-ziti-artifacts /tmp/ziti-edge-tunnel /usr/local/bin
    38c34,35
    < RUN chmod +x /docker-entrypoint.sh
    ---
    > COPY ./ziti-edge-tunnel /usr/local/bin/
    > RUN chmod +x /docker-entrypoint.sh ./usr/local/bin/ziti-edge-tunnel

I'm concerned that changes will be made to one dockerfile while the other is neglected.

Reply:
@scareything I think it would be OK to drop `Dockerfile.copy` based on my current estimation of the reduced likelihood that we'll need to continue building container images for older versions of `ziti-edge-tunnel` that pre-date the earliest published releases in GitHub.

Reply:
@scareything Re: quotes around TAG_PARAMS: that's correct. I used `eval` here just to avoid having a bare env var that's could be interpolated with arbitrary code as the value. I've been trying to be more intentional about quoting vars ever since I started using a shell script linter. It always points these out.

Reply:
@scareything Let me know if your can think of another way to introduce variations on a Dockerfile, but I've removed the almost-redundant `Dockerfile.copy` from this branch because I don't think we really need it.